src/fetch.ts

```diff
@@ -0,0 +1,101 @@
+/**
+ * Helpers for dealing with fetch requests
+ */
+
+import { TaskEither, tryCatch } from "fp-ts/lib/TaskEither";
+import { timeoutPromise } from "./promises";
+import { MaxRetries, RetriableTask, TransientError } from "./tasks";
+import { ITuple2, Tuple2 } from "./tuples";
+import { Millisecond } from "./units";
+
+/**
+ * An instance of fetch, along its associated AbortController
+ */
+export type AbortableFetch = ITuple2<typeof fetch, AbortController>;
+
+/**
+ * Makes fetch abortable and returns an AbortableFetch.
+ *
+ * The fetch request can be aborted by calling `AbortController`'s `abort()`.
+ *
+ * See https://developer.mozilla.org/en-US/docs/Web/API/AbortController/abort
+ */
+export function AbortableFetch(f: typeof fetch = fetch): AbortableFetch {
+  const abortController = new AbortController();
+  const signal = abortController.signal;
+
+  // augment the provided fetch with the signal controller
+  const abortableFetch: typeof fetch = (input, init) => {
+    const newInit = {
+      ...init,
+      signal
+    };
+    return f(input, newInit);
+  };
+
+  return Tuple2(abortableFetch, abortController);
+}
+
+/**
+ * Sets a timeout on an AbortableFetch.
+ *
+ * When the timeout expires, the controller's `abort()` method gets called.
+ */
+export function setFetchTimeout(
+  timeout: Millisecond,
+  abortableFetch: AbortableFetch
+): void {
+  // abort the fetch request on timeout
+  timeoutPromise(timeout)
+    .then(() => {
+      abortableFetch.e2.abort();
+    })
+    // tslint:disable-next-line:no-empty
+    .catch(() => {});
```

Comment:
i guess you can use void 0 instead of {} if you want to get rid of the tslint warning